File: frontend/src/services/checkin.ts
import api from './api'

export interface CheckinRecord {
  id: number
  checkin_date: string
  note?: string
  created_at: string
}

export interface CheckinStats {
  total_count: number
  current_streak: number
  longest_streak: number
  last_checkin_date: string | null
}

export interface CheckinResponse {
  message: string
  checkin: CheckinRecord
}

export interface CheckinListResponse {
  checkins: CheckinRecord[]
  pagination: {
    limit: number
    offset: number
  }
}

export interface CheckinStatsResponse {
  stats: CheckinStats
}

// 签到
export const doCheckin = async (note?: string): Promise<CheckinResponse> => {
  return await api.post('/checkin', { note })
}

// 获取签到记录
export const getCheckinRecords = async (
  limit: number = 30,
  offset: number = 0
): Promise<CheckinListResponse> => {
  return await api.get('/checkin', {
    params: { limit, offset },
  })
}

// 获取签到统计
export const getCheckinStats = async (): Promise<CheckinStatsResponse> => {
  return await api.get('/checkin/stats')
}

File: backend-python/app/schemas/checkin.py
from pydantic import BaseModel, Field
from datetime import datetime, date
from typing import Optional


class CheckinCreate(BaseModel):
    """签到创建请求"""
    note: Optional[str] = Field(None, max_length=500, description="签到备注")


class CheckinResponse(BaseModel):
    """签到响应"""
    id: int
    user_id: int
    checkin_date: date
    note: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


class CheckinStats(BaseModel):
    """签到统计响应"""
    total_count: int
    current_streak: int
    longest_streak: int
    last_checkin_date: Optional[date] = None
